src/app/Caches/Create/AdvancedOptions.tsx

```diff
@@ -0,0 +1,300 @@
+    const [isolationLevel, setIsolationLevel] = useState(props.advancedOptions.isolationLevel);
+    const [lockAcquisitionTimeout, setLockAcquisitionTimeout] = useState<number>(props.advancedOptions.lockAcquisitionTimeout || 0);
+    const [striping, setStriping] = useState<boolean>(props.advancedOptions.striping || true);
+
+    const [isOpenEvictionStrategy, setIsOpenEvictionStrategy] = useState(false);
+    const [isOpenIsolationLevel, setIsOpenIsolationLevel] = useState(false);
+    const [inputValueMaxSize, setInputValueMaxSize] = useState<number>(props.advancedOptions.maxSize);
+    const [inputValueMaxCount, setInputValueMaxCount] = useState<number>(props.advancedOptions.maxSize);
+
+    useEffect(() => {
+        props.advancedOptionsModifier({
+            storage: storage,
+            maxSize: maxSize,
+            maxCount: maxCount,
+            evictionStrategy: evictionStrategy,
+            concurrencyLevel: concurrencyLevel,
+            isolationLevel: isolationLevel,
+            lockAcquisitionTimeout: lockAcquisitionTimeout,
+            striping: striping,
+        });
+    }, [storage, maxSize, maxCount, evictionStrategy, concurrencyLevel, isolationLevel, lockAcquisitionTimeout, striping]);
+
+    const onToggleEvictionStrategy = () => {
+        setIsOpenEvictionStrategy(!isOpenEvictionStrategy);
+    };
+
+    const onSelectEvictionStrategy = (event, selection, isPlaceholder) => {
+        setEvictionStrategy(selection);
+        setIsOpenEvictionStrategy(false);
+        console.log('selected:', selection);
+    };
+
+    const onToggleIsolationLevel = () => {
+        setIsOpenIsolationLevel(!isOpenIsolationLevel);
+    };
+
+    const onSelectIsolationLevel = (event, selection, isPlaceholder) => {
+        setIsolationLevel(selection);
+        setIsOpenIsolationLevel(false);
+        console.log('selected:', selection);
+    };
+
+    const onToggleStriping = () => {
+        setStriping(!striping);
+    };
+
+    const handleConcurrencyLevel = (value) => {
+        setConcurrencyLevel(value);
+    }
+
+    const handleLockAcquisitionTimeout = (value) => {
+        setLockAcquisitionTimeout(value);
+    }
+
+    const onChangeMaxSize = (value, inputValue, setLocalInputValue) => {
+        let newValue;
+        if (inputValue === undefined) {
+            newValue = Math.floor(value);
+        } else {
+            if (inputValue > 100) {
+                newValue = 100;
+                setLocalInputValue(100);
+            } else if (inputValue < 0) {
+                newValue = 0;
+                setLocalInputValue(0);
+            } else {
+                newValue = Math.floor(inputValue);
+            }
+        }
+        setMaxSize(newValue);
+        setInputValueMaxSize(newValue)
+    };
+
+    const onChangeMaxCount = (value, inputValue, setLocalInputValue) => {
+        let newValue;
+        if (inputValue === undefined) {
```

Comment:
DRY with the method above

you can change this to !inputValue instead of === undefined